Changed graceful shutdown to use @godaddy/terminus

Bennoama opened the pull request.

Bennoama committed:
Adjusted to eslint

Bennoama committed:
eslint and prettier adjustments

Bennoama committed:
Merge branch 'main' into eslint-prettier-integration

Bennoama committed:
Removed unnecessary files

Bennoama committed:
Updated scripts to have format with prettier

Bennoama committed:
Changed graceful shutdown to use @godaddy/terminus

noamokman commented on src/framework/crud-db.ts:
```diff
@@ -4,6 +4,10 @@ import { client } from "../framework/mongo.js";
 import { formatDateInMMYYYY } from "../server/utils.js";
 
+export const isDBHealthy = async () => {
+  return (await client.db(dBName).command({ ping: 1 }))["ok"];
```
what is the return type here? boolean?

Bennoama replied:
number - 1 or 0
the command({ping:1}) returns object:
{
"ok" : val (0 or 1)
}

noamokman commented on src/framework/gracefulShutdown.ts:
```diff
@@ -1,12 +1,35 @@
+      await Promise.all([app.close(), mongo.disconnectDB()]);
+    },
+    healthChecks: {
+      "/healthcheck": async () => {
+        const response = {
+          dbHealthy: await isDBHealthy(),
+        };
+        return response;
+      },
```
this can be a one liner

noamokman commented on src/framework/gracefulShutdown.ts:
```diff
@@ -1,12 +1,35 @@
+      logger.info("Server shutting down");
+    },
+    onSignal: async () => {
+      await Promise.all([app.close(), mongo.disconnectDB()]);
```
you should actually wait for the server to close before disconnecting from the DB to not stop ongoing requests

noamokman commented on src/framework/gracefulShutdown.ts:
```diff
@@ -1,12 +1,35 @@
-    server.close();
-    mongo.disconnectDB();
-  };
+const loggerWrapper = (msg: string, err: Error) => {
+  if (err) {
+    logger.error(err.message);
+  } else {
+    logger.info(msg);
+  }
+};
```
you should pass the message/error to the extra properties and write your own message, not sure what logs are written here

Bennoama committed:
Fixed to PR review

1. Wait for server closing before disconnecting the db
2. healthcheck 1-liner
3. Logging moved to properties

Bennoama committed:
Merge branch 'main' into graceful-shutdown

Bennoama committed:
Changed import to include only dc function
